### PR TITLE
Local headmin merges shitty and broken code, puts up testmerge to resolve it.

### PR DESCRIPTION
**Pull request by Manly-Dwarf**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl: Manly Dwarf
fix: you know at first I thought I was being smug but then I realized the code didn't work at all, so the joke is on me. sleeping will now actually heal you again, assuming you aren't in deepcrit
fix: meaning that you should be fine if you're in softcrit, you can still sleep there
/:cl:

## About The Pull Request
imsorry

## Why It's Good For The Game
imsorry

Don't merge this until I'm awake and can remove the testmerge.

### CODE REVIEW

**Comment by ghost** [comment]
#### Test Merge Deployed

##### Server Instance
HippieStation

##### Revision
Origin: d8d64afff91595d10522c1bb713bd33573fe9b61
Pull Request: 45e395fed76204003ee8fc7a2b6b9e10626fc58e
Server: 25510209971defcf93e0c1a2ac64f014ec42ffbb

##### Comment
for shame

**Comment by mikolaj007** [comment]
+1. while it is important to double-tap your corpses, nobody should get a "get out of jail free" card on hardcrit

**Comment by ghost** [comment]
does sleeping out of _soft_-crit still work?

**Comment by ghost** [comment]
actually, looking at the code...
-1 it doesn't.
i'd prefer the check be
```owner.health > HEALTH_THRESHOLD_FULLCRIT```

**Comment by mikolaj007** [comment]
oh yea, got wrong number
+1 if its HEALTH_THRESHOLD_FULLCRIT instead of crit

**Comment by Manly-Dwarf** [comment]
What do you mean, steamp0rt?
It worked as it should when I tested it. No healing while unconscious, no healing while crit, but healing while not crit.

**Comment by ghost** [comment]
@Manly-Dwarf i'd prefer if you could heal out of softcrit (where you can crawl), but not hard-crit (where you're KO'd)

**Comment by Manly-Dwarf** [comment]
bro, no
first off this was a code oversight. I even said originally in the PR that I only approved of it if healing on crit was checked on, and it wasn't.

second off, why should you be able to heal out of crit just because? we discussed this in the designer channel, it's goes against the entire purpose of crit, in that you're helpless and at the mercy of the crew. just being to heal out is stupid, your time for healing has come and gone, as has using the radio. if you didn't inject some chems beforehand, don't have implants, or don't have a pal then there's a decent chance that you're screwed.

**Comment by ghost** [comment]
> second off, why should you be able to heal out of crit just because? we discussed this in the designer channel, it's goes against the entire purpose of crit, in that you're helpless and at the mercy of the crew. just being to heal out is stupid, your time for healing has come and gone, as has using the radio. if you didn't inject some chems beforehand, don't have implants, or don't have a pal then there's a decent chance that you're screwed.

Ok, i'm going to use an example. Let's say you have a holoparasite. Some nerd is chasing you in maint. That nerd softcrits you, but the holopara gives them the slip and gets you to a safe place. Now what? You're perma-dead because some idiot jumped you with dual-wielded lasers and you forgot a burn patch? Sounds stupid if you ask me.

**Comment by Manly-Dwarf** [comment]
I mean that's just how it has always worked. This is not a solution to that if you ask me.
It's unfortunate when you get critted and you're helpless, but it's a rule that everyone works under, even holoparasite users, and is the point of crit to begin with.

If we want to make combat more lethal, we can't make lethals worse, and this kinda does that

**Comment by ghost** [comment]
What's the point of softcrit, then?

**Comment by Manly-Dwarf** [comment]
Crawling and whispering.

**Comment by Manly-Dwarf** [comment]
The point of unconscious crit is whispering "FUCK YOU" so it can be your last dying breath.

**Comment by NautilusViper** [comment]
I am going to cuckold Manly Dwarf

**Comment by Manly-Dwarf** [comment]
who is this nautilusviper?

File: hippiestation/code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -19,7 +18,7 @@
 
 
 /datum/status_effect/incapacitating/sleeping/tick()
-	if(!owner.stat > sleep_stat_allowed)
+	if(owner.health > owner.crit_threshold) // shittier code but at least it verified works.
```
Comment (NautilusViper) [inline hippiestation/code/datums/status_effects/debuffs.dm]: HEALTH_THRESHOLD_FULLCRIT because literally nobody ever complained about this

Comment (Manly-Dwarf) [reply hippiestation/code/datums/status_effects/debuffs.dm]: 2bad

**Comment by Manly-Dwarf** [comment]
"nobody complained about this" a good feature does not make. nobody complained about flashes for a long time, nobody complained about riot shields, nobody complained about armory hoarding, until they realized how dumb it was because it got abused

**Comment by NautilusViper** [comment]
> first off this was a code oversight

for the better

> second off, why should you be able to heal out of crit just because?

because the threshold between softcrit and fullcrit is 30 damage, or two lit welder hits

> if you didn't inject some chems beforehand, don't have implants, or don't have a pal then there's a decent chance that you're screwed

hmmm today I will encourage powergaming through code (clueless)

> I mean that's just how it has always worked

this argument gives me a fucking ulcer luigi

> It's unfortunate when you get critted and you're helpless, but it's a rule that everyone works under, even holoparasite users, and is the point of crit to begin with.

and being able to sleep off lightcrit is also a rule that everyone would be able to work under, including holoparasite users. being blacked out and unable to move is what I think of when I hear 'critical condition', lightcrit is basically just getting incapacitated in left 4 dead or going into bleedout in payday, it's your 'down but not out' state and it makes no sense to be completely SOL

> If we want to make combat more lethal, we can't make lethals worse, and this kinda does that

I thought what 'we' wanted was less stun bullshit and more ballistics/stamina-based combat but okay. I would fix the good/bad/grey PR and get it merged before touching this.

>nobody complained about riot shields, nobody complained about armory hoarding

PLEASE say sike this is the kind of shit people bitch about nonstop in saltchat

**Comment by Manly-Dwarf** [comment]
>for the better

nah, like I said an argument could be made for softcrit but not healing yourself out of the brink of death just because, crit is a state of helplessness and should remain that way.

>threshold

it's still crit, and still a state of helplessness.

the rest of this is salt so I'll address the main argument
>and being able to sleep off lightcrit is also a rule that everyone would be able to work under, including holoparasite users. being blacked out and unable to move is what I think of when I hear 'critical condition', lightcrit is basically just getting incapacitated in left 4 dead or going into bleedout in payday, it's your 'down but not out' state and it makes no sense to be completely SOL

Crit used to operate entirely on blackout, you got below 0 HP and bam you were fucked. Softcrit and whispering came along, then crawling.
Every action you make in softcrit has a cost. Whispering and crawling drain your stamina. Crawling also makes you track blood, which increases your bloodloss. Healing does not have an inherent cost other than "not moving/talking", two things that actually have a cost, and all things considered type *deathgasp first and you'll probably trick the guy killing you anyhow if they're not smart.

Crit is not "down but not out". That's how it works on goon, crit there just slows you down a fuckton, fucks with your speech, and makes you a ticking timebomb until you drop out for good. Pass a certain damage threshold and you're permastunned, and usually when you pass that threshold it's because you've taken 300 oxygen, and you're just waiting for the game to run out a timer and kill you. There's also other systems like multi-organ failure, which is the REAL killer on Goon.
Crit on /tg/code is completely different. Once you hit the damage threshold you're on the ground, unable to shoot, unable to fight back, and can only talk or desperately crawl away. As I mentioned previously, these all have a price to pay, so I fail to see why just going to sleep to heal yourself is justified when the risks are barely any different from regular crit.
There's no organ failure, there's bloodloss, which is generally quite slow, and oxygenloss, which also takes some time to kick in. These generally won't stop you from healing yourself out of critical, no matter how far gone you are. The argument for softcrit is that there's a small difference between that and "completely fucked", and I would be inclined to agree, but it's still too obvious of a choice. Would you rather desperately crawl away and get the guy to smack you to death or try to get to medical if you're in the halls and HOPE someone heals you, beg for your life, or just play dead and heal yourself?

One thing I did NOT mind was slamming epinephrine into you before you got crit to heal yourself. That was far more tolerable since it required some form of quick thinking and effort, rather than just pushing the sleep button which makes you helpless when you're already helpless. If you seriously want healing while crit back, you should revert the epinephrine /tg/ nerf.

**Comment by ghost** [comment]
#### Test Merge Updated

##### Server Instance
HippieStation

##### Revision
Origin: 15b014c2c143c675564359292aafb5221eb29454
Pull Request: 45e395fed76204003ee8fc7a2b6b9e10626fc58e
Server: f5c3a9e7480a27a8a90dbbf81e8923a4a6063014

##### Comment
for shame

**Comment by NautilusViper** [comment]
> nah, like I said an argument could be made for softcrit but not healing yourself out of the brink of death just because, crit is a state of helplessness and should remain that way.

yeah........................................................................................................................................ we agree

> it's still crit, and still a state of helplessness.

it's reduced helplessness but you aren't at the complete mercy of the elements until fullcrit happens, which is, as I said, not all that far off from softcrit

> the rest of this is salt so I'll address the main argument

nah, i'll run through them again so you can address them: appeal to tradition isn't a defense or an argument, 'it was that way before' is a totally empty statement. saying "you should've injected chems/had a chem implant/had a buddy up your ass to save you from trouble" is defending shitty powergaming that nobody likes, sleeping to heal out of (soft)crit is a way for anyone to potentially save themselves from being forced out of the round permanently without resorting to the same 'rush ghettochem and make heals' tactic every round. I don't know who said "we want to make combat more lethal" but it's news to me. riot shields and the ol' armory-in-the-backpack are both fairly spotlight topics.

> Crit used to operate entirely on blackout, you got below 0 HP and bam you were fucked. Softcrit and whispering came along, then crawling.
Every action you make in softcrit has a cost. Whispering and crawling drain your stamina. Crawling also makes you track blood, which increases your bloodloss. Healing does not have an inherent cost other than "not moving/talking", two things that actually have a cost, and all things considered type *deathgasp first and you'll probably trick the guy killing you anyhow if they're not smart.

sure, that was the way things may have been, but the way they are now are much better. sleeping effectively sends you into the same state of complete helplessness as deepcrit and has a bit of flavor text to give the fact that you ARE sleeping away upon examine to be 100% sure. It takes away your ability to crawl to safety, so if you sleep near space, for example, instead of crawling away and then sleeping, you're totally fucked. tricking dumb people into mistakenly thinking they killed you and getting away with it is a good thing in every possible sense of the term.

> Crit is not "down but not out"

SOFTcrit should be. that's my point.

> The argument for softcrit is that there's a small difference between that and "completely fucked", and I would be inclined to agree, but it's still too obvious of a choice. Would you rather desperately crawl away and get the guy to smack you to death or try to get to medical if you're in the halls and HOPE someone heals you, beg for your life, or just play dead and heal yourself?

sure, four times out of five you'll go with the sleep route, but what if you've already lost a lot of blood and are still bleeding? what if you're near open space? what if medbay is excruciatingly close and you KNOW there's competent medical staff on hand (rare, but happens)? all of these are valid reasons to choose 'move' instead of 'stay sedentary and hope you heal back to at least blinking red before something kills you'.

> One thing I did NOT mind was slamming epinephrine into you before you got crit to heal yourself. That was far more tolerable since it required some form of quick thinking and effort, rather than just pushing the sleep button which makes you helpless when you're already helpless. If you seriously want healing while crit back, you should revert the epinephrine /tg/ nerf.

I said this in OOC, but frankly just replace epinephrine pens with atropine pens, they have robust as fuck crit-exclusive healing which is perfect for saving lives but they're bad for combat due to causing screen shake.

**Comment by yoyobatty** [comment]
What a shit show, is this ready?

**Comment by Manly-Dwarf** [comment]
Nope, I find myself convinced. I'll update the PR to let you heal in softcrit again.

**Comment by NautilusViper** [comment]
> Nope, I find myself convinced. I'll update the PR to let you heal in softcrit again.

**Comment by ghost** [comment]
#### Test Merge Updated

##### Server Instance
HippieStation

##### Revision
Origin: 9bf2c62463c3bd5b4f5062016a66dff8da27d223
Pull Request: 45e395fed76204003ee8fc7a2b6b9e10626fc58e
Server: 1a4c29f1d287e617243c0030809ebb3399dfaf94

##### Comment
if honkbot posts anything it's because honkbot is stupid

**Comment by ghost** [comment]
#### Test Merge Updated

##### Server Instance
HippieStation

##### Revision
Origin: 9bf2c62463c3bd5b4f5062016a66dff8da27d223
Pull Request: 45e395fed76204003ee8fc7a2b6b9e10626fc58e
Server: ccfb70a20c712d40a9d2a14d17c13b8c337b83dc

##### Comment
if honkbot posts anything it's because honkbot is stupid

**Comment by Manly-Dwarf** [comment]
Testmerging the fullcrit variant.

**Comment by ghost** [comment]
#### Test Merge Updated

##### Server Instance
HippieStation

##### Revision
Origin: 72e72e24825f69dc8cfa86a36db3a9577043fe3e
Pull Request: 45e395fed76204003ee8fc7a2b6b9e10626fc58e
Server: 4c659a75acdb4032c0951b768d9d2dd50e78b2c9

##### Comment
fullcrit

**Comment by ghost** [comment]
#### Test Merge Updated

##### Server Instance
HippieStation

##### Revision
Origin: 72e72e24825f69dc8cfa86a36db3a9577043fe3e
Pull Request: beb431ed2bfc31a4d22c9a2690f078ab86e68aa4
Server: 472ce7fa8d70296e12e68c175967b9cf10a42994

##### Comment
fullcrit

**Comment by ghost** [comment]
#### Test Merge Updated

##### Server Instance
HippieStation

##### Revision
Origin: 4705fd9e7ae76fea77ac58bc62456e73152df7f4
Pull Request: beb431ed2bfc31a4d22c9a2690f078ab86e68aa4
Server: ae9a7a0a76f9d9fece3506fe2efa31e37e6a80f9

##### Comment
fullcrit

**Comment by AnthonyJNgo** [comment]
This has my +1 by the way, for either versions of it.

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
alright well the code looks fine and it's headmin approved, so ima merge it

**Comment by ghost** [comment]
#### Test Merge Removed

**Pull request merged**